datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -1439,6 +1441,57 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_statistics_from_parquet_metadata_dictionary() -> Result<()> {
+        // Data for column c_dic: ["a", "b", "c", "d"]
+        let values = StringArray::from_iter_values(["a", "b", "c", "d"]);
+        let keys = Int32Array::from_iter_values([0, 0, 1, 2]);
+        let dic_array =
+            DictionaryArray::<Int32Type>::try_new(keys, Arc::new(values)).unwrap();
+        let boxed_array: Box<dyn arrow_array::Array> = Box::new(dic_array);
+        let c_dic: ArrayRef = Arc::from(boxed_array);
+
+        // Define the schema
+        let field = Field::new(
+            "c_dic",
+            DataType::Dictionary(Box::new(DataType::Int32), Box::new(DataType::Utf8)),
+            false,
+        );
+        let schema = Schema::new(vec![field]);
+        // Create the RecordBatch
+        let batch1 = RecordBatch::try_new(Arc::new(schema), vec![c_dic]).unwrap();
```

Comment:
You can probably make this much shorter by using https://docs.rs/arrow/latest/arrow/record_batch/struct.RecordBatch.html#method.try_from_iter

Here is an example: https://github.com/apache/datafusion/blob/7a7797c891057c783680bade813d4ba2a49344b7/datafusion-examples/examples/expr_api.rs#L116

The benefit is that it will make the schema for you from the `ArrayRef`s (which already know their type)

There is also an example of this earlier in this file that might help https://github.com/apache/datafusion/pull/11169/files#diff-0761860da9ee8df947358b2589079cbf64690c6228cc27dd41a21fab27400f75R1361

Reply:
Thanks for this - I think it will also turn out to be useful in simplifying some other tests I'm writing.